lib/dealership.rb

```diff
@@ -0,0 +1,60 @@
+class Dealership
+  attr_reader :name, :address, :inventory
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+
+  end
+
+  def inventory_count
+    @inventory.count
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    if @inventory.count > 0
+      true
+    else
+      false
+    end
+  end
+
+  def cars_by_make(make)
+    cars_by_make = []
+     @inventory.each do |car|
+       if car.make == make
+         cars_by_make << car
+        end
+     end
+     return cars_by_make
+  end
+
+  def total_value
+    total_value = 0
+    @inventory.each do |car|
+      total_value += car.total_cost
+    end
+    total_value
+  end
+  def details
+    details = {
+      "total_value" => total_value,
+      "address" => @address
+    }
+  end
+
+  def average_price_of_car
+      average = total_value / inventory_count
+      return average
+  end
+
+  def cars_sorted_by_price
+    @inventory.each do |sort|
+      inventory.sort_by(sort.total_cost)
+    end
+  end
```

Comment:
This is interesting. You're using two different enumerables here: `each` and `sort_by`. What is your expectation of what would happen here, and what was the actual output?